pulumi/aws/docs/getting_started.md

```diff
@@ -0,0 +1,177 @@
+    git, docker, make).
+ 3. Setup AWS credentials.
+ 4. Setup Pulumi credentials.
+ 5. Run `start_all.sh` and answer the prompts.
+
+## Install on MacOS with HomeBrew and Docker Desktop
+```
+# Install Homebrew for the Mac: https://brew.sh/
+# Install Docker Toolbox for the Mac: https://docs.docker.com/docker-for-mac/install/
+$ brew install make git python3
+```
+
+## Install on MacOS with Docker Desktop
+```
+# In a terminal window with the MacOS UI, install developer tools if they haven't already 
+# been installed.
+$ xcode-select --install
+$ bash ./setup_venv.sh
+```
+
+## Install with Debian/Ubuntu Linux
+```
+$ sudo apt-get update
+$ sudo apt-get install --no-install-recommends curl ca-certificates git make python3-venv docker.io
+$ sudo usermod -aG docker $USER
+$ newgrp docker
+$ bash ./setup_venv.sh
+```
+
+## Install with CentOS/Redhat Linux
+```
+# Install Docker Yum repository
+$ sudo dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
+$ sudo yum install python3-pip make git docker-ce
+$ sudo systemctl enable --now docker
+$ sudo usermod -aG docker $USER
+$ newgrp docker
+$ bash ./setup_venv.sh
+```
+
+## Creating a Debian Docker Runtime Environment
+
+Run the following helper script to build a Debian container image.
+```
+$ ./build_dev_docker_image.sh debian
+```
+
+## Creating a CentOS Docker Runtime Environment 
+
+Run the following helper script to build a CentOS container image.
+```
+$ ./build_dev_docker_image.sh centos
+```
+
+## Stand Alone Install
+
+### Requirements
+
+#### Python 3 or Prerequisites for Building Python
+
+In this project, Pulumi executes Python code that creates cloud and Kubernetes
+infrastructure. In order for it to work, Python 3 and the [venv module](https://docs.python.org/3/library/venv.html) 
+must be installed. Alternative, if GNU make and the gcc compiler are installed
+the setup script can build and install Python 3.
+
+#### Git
+
+The `git` command line tool is required for checking out KIC source code from
+github and for the KIC image build process.
+
+#### Make
+
+In order to build the Ingress Controller from source, GNU `make` is required
+to be installed on the running system.
+
+#### Docker
+
+Docker is required because the Ingress Controller is a Docker image and needs
+Docker to generate the image.
+
+#### Kubernetes
+
+Although not required, installing the [CLI tool `kubectl`](https://kubernetes.io/docs/tasks/tools/)
+will allow you to interact with the Kubernetes cluster that you have stood up
+using this project.
+
+#### Setup
+
+Within the project, you will need to install Python and dependent libraries 
+into the `venv` directory. To do this is to invoke the [`setup_venv.sh`](./setup_venv.sh) 
+included in the project. This script will install into the [virtual environment](https://docs.python.org/3/tutorial/venv.html)
+directory:
+
+ * Python 3 (via pyenv) if it is not already present
+ * Pulumi CLI utilities
+ * AWS CLI utilities 
```

Comment:
The only thing I would note here (since it confused me) is that the version of the AWS CLI that we install with this command is on the 1.x release line (versus the 2.x that is likely to be installed in the OS).

Reply:
Yes, unfortunately the v2 AWS client can't be installed via PIP and it is like 300mb. We could switch to it at some point, but it seems like v1 works and embeds more cleanly.

Reply:
That's where I got to when I was digging - also, we're not really doing anything crazy with the client so the lighter one works just fine. The only reason I would consider a switch is if we either needed features of the updated version or if v1 stopped getting patched/supported.